templates/web/base/auth/generate_token.html

```diff
@@ -0,0 +1,50 @@
+[%
+SET bclass = 'authpage';
+SET bclass = 'fullwidthpage' IF password_changed;
+INCLUDE 'header.html', title = loc('Change password'), bodyclass = bclass
+%]
+
+[% IF token_generated %]
+
+  <div class="confirmation-header">
+    <h1>[% loc('Your token has been generated') %]</h1>
+
+    <div class="form-field">
+        <strong>[% loc('Token:') %]</strong>
+        <span>[% existing_token | html %]</span>
+    </div>
+
+    <p><a href="/my">[% loc('Your account') %]</a></p>
+  </div>
+
+[% ELSE %]
+
+<h1>[% loc('Generate Token') %]</h1>
+
+<form action="[% c.uri_for_action('/auth/profile/generate_token') %]" method="post" name="generate_token" class="fieldset">
+    <input type="hidden" name="token" value="[% csrf_token %]">
+
+    <fieldset>
+
+    [% IF existing_token %]
+        <div class="form-field">
+            <strong>[% loc('Token:') %]</strong>
+            <span>[% existing_token | html %]</span>
+        </div>
+
+        <div>
+          <p>
+          [% loc('If you generate a new token the existing token will no longer work.') %]
```

Comment:
I'd put this message underneath the button (as it is, it looks like it's part of the token display).